db/compaction/compaction_job.cc

```diff
@@ -250,6 +250,14 @@ void CompactionJob::Prepare() {
           (i != boundaries_.size()) ? std::optional<Slice>(boundaries_[i])
                                     : std::nullopt,
           static_cast<uint32_t>(i));
+
+      size_t ts_size = cfd->user_comparator()->timestamp_size();
+      (void)ts_size;
+      assert(i == 0 || i == boundaries_.size() ||
+             cfd->user_comparator()->CompareWithoutTimestamp(
+                 StripTimestampFromUserKey(boundaries_[i - 1], ts_size), false,
+                 StripTimestampFromUserKey(boundaries_[i], ts_size),
+                 false) != 0);
```

Comment:
This can be
```
cfd->user_comparator()->CompareWithoutTimestamp(
    boundaries_[i-1], true, boundaries_[i], true) < 0
```

Reply:
Oh. I thought that would fail in case of timestamp disabled flow as comparator expects keys to have timestamps.